analyses/ZplusJet_3.cc

```diff
@@ -107,103 +132,147 @@ namespace Rivet {
 
       // Retrieve dressed leptons, sorted by pT
       vector<DressedLepton> leptons = apply<DressedLeptons>(event, "leptons").dressedLeptons();
-      MSG_DEBUG("Lepton multiplicity = " << leptons.size());
 
-      // discard events with less than two muons
+      // discard events with less than two and more than maximum number of leptons
       if (leptons.size() < 2) vetoEvent;
+      if (leptons.size() > _maxnleptons) vetoEvent;
+      MSG_DEBUG("Found " << leptons.size() << " leptons");
+      for (auto lep: leptons) {
+        MSG_DEBUG("\tlepton pt: " << lep.pT());
+        MSG_DEBUG("\tlepton y: " << lep.rap());
+      }
 
       // Retrieve clustered jets, sorted by pT, with a minimum pT cut
-      Jets jets = apply<FastJets>(event, "jets").jetsByPt(Cuts::absrap < 2.4 && Cuts::pT > 10*GeV);
+      map<string,Jets> _jetcollections;
+      _jetcollections["AK4"] = apply<FastJets>(event, "jetsAK4").jetsByPt(Cuts::absrap < _maxabsjetrap && Cuts::pT > _jetpt);
+      _jetcollections["AK8"] = apply<FastJets>(event, "jetsAK8").jetsByPt(Cuts::absrap < _maxabsjetrap && Cuts::pT > _jetpt);
+
+      // Require at least one jet in any jet collection with a minimum pT 
+      bool jet1pass = false;
+
+      set<string> _jetcollectionstoerase;
+
+      for (auto& jets: _jetcollections) {
+        // Remove all jets within dR < 0.3 of a dressed lepton
+        idiscardIfAnyDeltaRLess(jets.second, leptons, 0.3);
+        MSG_DEBUG("After lepton cleaning jet multiplicity " << jets.first << "= " << jets.second.size());
+        for (auto jet: jets.second) {
+          MSG_DEBUG("\tjet pt: " << jet.pT()/GeV);
+          MSG_DEBUG("\tjet y: " << jet.rap());
+        }
+
+        // Require at least one hard jet
+        if (!jets.second.empty()) {
+          if (jets.second.at(0).pT() > _minjet1pt) {
+            MSG_DEBUG("Hardest " << jets.first << " jet pt: " << jets.second.at(0).pT());
+            jet1pass = true;
+          } else {
+            _jetcollectionstoerase.insert(jets.first);
+          }
+        }
+        else {
+            _jetcollectionstoerase.insert(jets.first);
+        }
+      }
 
-      // Remove all jets within dR < 0.3 of a dressed lepton
-      idiscardIfAnyDeltaRLess(jets, leptons, 0.3);
-      MSG_DEBUG("Jet multiplicity = " << jets.size());
+      if (!(jet1pass)) vetoEvent;
 
-      // Require at least one hard jet
-      if (jets.empty()) vetoEvent;      
-      if (jets.at(0).pT() <= 20*GeV) vetoEvent;
+      for (auto c: _jetcollectionstoerase) {
```

Comment:
is c a pointer? I think it should be, but better to be sure that the collection is actually erased.